agent/retry.go

```diff
@@ -0,0 +1,46 @@
+package agent
+
+import (
+	"errors"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/odpf/meteor/plugins"
+)
+
+const (
+	defaultRetryTimes      = 5
+	defaultInitialInterval = 5 * time.Second
+)
+
+func retryIfNeeded(operation func() error, retryTimes int, initialInterval time.Duration, notify func(e error, d time.Duration)) error {
```

Comment:
@StewartJingga How about we make retry interface to something like. 
```
attempt := retrier.new(interval, maxRetries)
attempt.make(fn)
```
We can find better names though. Wdyt?

Reply:
@ravisuhag  was trying to make it into a struct before.
but I was waiting for it to be more complex to justify it. wdyt?

Reply:
Readability is a challenge right now. So I would suggest that we can do it now.